storage/pruning/pruningStorer.go

```diff
@@ -139,16 +128,62 @@ func initPruningStorer(
 		pdb.bloomFilter = bf
 	}
 
-	err = pdb.activePersisters[0].persister.Init()
-	if err != nil {
-		return nil, err
-	}
-
 	pdb.registerHandler(args.Notifier)
 
 	return pdb, nil
 }
 
+func initPersistersInEpoch(
+	args *StorerArgs,
+	shardIdStr string,
+) ([]*persisterData, map[uint32]*persisterData, error) {
+	var persisters []*persisterData
+	persistersMapByEpoch := make(map[uint32]*persisterData)
+
+	if !args.PruningEnabled {
+		p, err := createPersisterDataForEpoch(args, 0, shardIdStr)
+		if err != nil {
+			return nil, nil, err
+		}
+		persisters = append(persisters, p)
+		return persisters, persistersMapByEpoch, nil
+	}
+
+	if args.NumOfEpochsToKeep < args.NumOfActivePersisters {
```

Comment:
can these be different? The NumOfEpochsToKeep and the NumOfActivePersisters?

Reply:
yes, we can persist more epochs data, but the active persisters are the number of most recent persisted epochs that are also opened and can have their data directly accessed.

Reply:
ok